document-portal/document-portal.c

```diff
@@ -1416,6 +1416,68 @@ portal_list (GDBusMethodInvocation *invocation,
   return TRUE;
 }
 
+static gboolean
+portal_get_real_path (GDBusMethodInvocation *invocation,
+             GVariant *parameters,
+             XdpAppInfo *app_info)
+{
+  const char *id = NULL;
+  g_autoptr(PermissionDbEntry) entry = NULL;
+
+  g_variant_get (parameters, "(&s)", &id);
+
+  XDP_AUTOLOCK (db);
+
+  entry = permission_db_lookup (db, id);
+
+  if (!entry)
+    {
+      g_dbus_method_invocation_return_error (invocation,
+                                             XDG_DESKTOP_PORTAL_ERROR, XDG_DESKTOP_PORTAL_ERROR_INVALID_ARGUMENT,
+                                             "Invalid ID passed");
+      return TRUE;
+    }
+
+  if (!xdp_app_info_is_host (app_info))
+    {
+      g_autofree const char **apps = NULL;
+      g_autofree const char *app_id = NULL;
```

Comment:
This will cause double-free. You don't have to free `app_id` in this case.

Reply:
I haven't checked, but this is most likely the reason why the test fails. It was also not working for me as it crashed on double-free and removing the `g_autofree` made it work.

Reply:
I also don't see the full log, but running the test locally with asan, it doesn't pass, because there is a memory leak.

This works for me:
```
diff --git a/tests/test-doc-portal.c b/tests/test-doc-portal.c
index e2d9f0c..cd39912 100644
--- a/tests/test-doc-portal.c
+++ b/tests/test-doc-portal.c
@@ -687,16 +687,11 @@ test_add_named (void)
   assert_doc_not_exist (id1, basename1, "com.test.App2");
 }
 
-static char *
-doc_get_real_path (const char *basename)
-{
-  return g_build_filename (outdir, basename, NULL);
-}
-
 static void
 test_get_real_path (void)
 {
   g_autofree char *doc_id = NULL;
+  g_autofree char *expected_real_path = NULL;
   g_autofree char *real_path = NULL;
   const char *basename = "real-path";
   g_autoptr(GVariant) reply = NULL;
@@ -720,7 +715,8 @@ test_get_real_path (void)
 
   g_assert_no_error (error);
   g_variant_get (reply, "(s)", &real_path);
-  g_assert_cmpstr (real_path, ==, doc_get_real_path(basename));
+  expected_real_path = g_build_filename (outdir, basename, NULL);
+  g_assert_cmpstr (real_path, ==, expected_real_path);
 }
 
 static void
```

Reply:
Can you please remove the static function? I don't think there is a need for it as it's just one-liner and it's not used anywhere else. Also please squash your commits into one.

Reply:
Aha, sorry, you did that already, I was just looking at the last commit.